proposals/044-integration-model.md

```diff
@@ -0,0 +1,203 @@
+* Exactly one of these integrations can be active.
+* Each deployment of an Integration creates a new version.
+* Users can switch integration versions from the UI
+* Versioned Integrations can be inspected from a history page.
+* Old integrations can't be changed but can be used as a blueprint for new integrations.
+* As soon as an integration is changed, a new version of an integration is created. All changes go to this new integration until it is deployed by the user. After this, its gets a fixed version and cannot be changed anymore.
+* The user must always be able to see and examine the currently active integration along with its version number.
+
+_Attention: THIS IS WORK-IN-PROGRESS. The propsoal is PRed early to collect early feedback_
+
+
+## Integration state model
+
+Technical an Integration is the domain object describing a Camel Route. 
+It is managed by `syndesis-rest` and `syndesis-ui`. 
+During design time this object only lives in this domain. 
+After a "Publish" operation the integration is transformed in an runnable artefact. 
+This runnable artefact consists of a GitHub repo containing the code and configuration and an OpenShift resource configuration which transforms this code into a running application, backed by an integration pod.
+Running integrations can be suspended and resumed without destroying the deployment.
+
+
+![Integration state model](images/integration-state.png "Integration State Model")
+
+| State       | Description |
+| ----------- |------------ |
+| Draft | Initial state of an integration. The integration is not yet deployed |
+| Active | Integration is deployed and running |
+| Inactive | Integration is deployed but is not running|
+| Undeployed | An integration has been undeployed | 
+| Error | The integration is deployed but in an error state |
+
+### Draft
+
+Right after an integration is created it enters the _Draft_ state. This means 
+the integration is not yet deployed. An integration leaves the draft state either by being published or deleted (in which case the whole integration is removed)
+
+After an activation there is no way back to the _Draft_ state.
+
+### Active
+
+An _active_ integration is a running integration. This state can be either created from a _Draft_ state when the integration gets published of from an _Inactive_ state when a suspended integration is started again.
+
+### Inactive
+
+An _inactive_ integration is an integration which was previously active and has been suspended by some action. An _inactive_ integration can be resumed or undeployed
+
+### Undeployed
+
+An integration which was published is never deleted but can be undeployed. An undeployed integration can be re-deployed (and activated) if the user chose the version of this integration to be reactivated.
+
+### Error
+
+The error state indicates that the integration is not running. This state can be entered consciously but only as a side effect. An integration can enter the error state any time. 
```

Comment:
If possible I'd like to keep a single _Error_ state to keep it simple. This state can be left to _Active_ if the error is transient and can get fixed. Along with this state goes a `stateDescription` which can hold detailed information about the error, maybe with some hint how to fix it.

"Error" is not necessarily a final state.